gwpy/table/io/gstlal.py

```diff
@@ -0,0 +1,164 @@
+
+    See also
+    --------
+    gwpy.io.ligolw.read_table
+        for details of keyword arguments for the read operation
+    gwpy.table.io.ligolw.to_astropy_table
+        for details of keyword arguments for the conversion operation
+    """
+
+    extra_cols = []
+    if triggers == 'single':
+        derived_cols = []
+        if kwargs['columns']:
+            for name in kwargs['columns']:
+                if name not in lsctables.TableByName['sngl_inspiral'].validcolum:
+                    if name in GET_COLUMN:
+                        derived_cols.append(name)
+                        kwargs['columns'].pop(name)
+                        required_cols = GET_COLUMN_EXTRA[name]
+                        for r_col in required_col if r_col not in kwargs['columns']:
+                            kwargs['columns'].append(r_col)
+                            extra_cols.append(r_col)
+                    else:
+                        raise
+        events = read_table(source, tablename='sngl_inspiral', **kwargs)
+        for col_name in derived_cols:
+            col_data = GET_COLUMN[col_name](events)
+            events.add_column(col_data,name=col_name)
+    if triggers == 'coinc':
+        if kwargs['columns']:
+            columns = kwargs['columns']
+            kwargs.pop('columns')
+            # Divvy up columnss
+            if 'coinc_event_id' not in columns:
+                columns.append('coinc_event_id')
+                extra_cols.append('coinc_event_id')
+            inspiral_cols = [col if col in lsctables.TableByName['coinc_inspiral'].validcolumn
+                             for col in columns]]
+            event_cols = [col if col in lsctables.TableByName['coinc_event'].validcolumn
+                             for col in columns]]
+            if 'end' in columns:
+                inspiral_cols.append('end')
+            coinc_inspiral = read_table(source, tablename='coinc_inspiral', columns=inspiral_cols, **kwargs)
+            coinc_event = read_table(source, tablename='coinc_event', columns=event_cols, **kwargs)
+        else:
+            coinc_inspiral = read_table(source, tablename='coinc_inspiral', **kwargs)
+            coinc_event = read_table(source, tablename='coinc_event', **kwargs)
+        events = join(coinc_inspiral, coinc_event, keys="coinc_event_id",
+                      metadata_conflicts='silent')
+        events.meta['tablename'] = 'gstlal_coinc_inspiral'
+    else:
+        raise
+    for col_name in extra_cols::
+        events = events.remove_column(col_name)
+    return events
+
+
+### !!! unclear how to set up register
+def identify_gstlal(origin, filepath, fileobj, *args, **kwargs):
+    """Identify a PyCBC Live file as an HDF5 with the correct name
+    """
+    if identify_hdf5(origin, filepath, fileobj, *args, **kwargs) and (
+            filepath is not None and PYCBC_FILENAME.match(basename(filepath))):
+        return True
+    return False
```

Comment:
@dethodav, the `identify_gstlal` function should take in the arguments are you have them listed, and inspect them to unimbiguously detect whether the incoming file is a 'gstlal file' (returning `True`) or not (`False`).

For PyCBC we do this with a regular expression on the filename, which isn't great, but seems to work well enough.

What is unique about a GstLAL file?